### ISSUE TITLE
Add `grpc reflection` for tools like grpcui.

### ISSUE THREAD

**Issue by hyeonjae** [argoproj/argo#2879]
# Summary

Add `grpc reflection` for tools like grpcui. 😄 

# Motivation

I want to call argo api using tool like [grpcui](https://github.com/fullstorydev/grpcui) (similar with postman) which lets us interect with grpc server via a browser.
For this, the grpc reflection need to be added.  🙏 

# Proposal

Add only 2 lines of code in `argoserver.go`

```diff
index 0db3d03c..d58ec8af 100644
--- a/server/apiserver/argoserver.go
+++ b/server/apiserver/argoserver.go
@@ -15,6 +15,7 @@ import (
        "golang.org/x/net/context"
        "google.golang.org/grpc"
        "google.golang.org/grpc/credentials"
+       "google.golang.org/grpc/reflection"
        "k8s.io/apimachinery/pkg/util/wait"
        "k8s.io/client-go/kubernetes"
        "k8s.io/client-go/rest"
@@ -213,6 +214,7 @@ func (as *argoServer) newGRPCServer(instanceID string, offloadNodeStatusRepo sql
        cronworkflowpkg.RegisterCronWorkflowServiceServer(grpcServer, cronworkflow.NewCronWorkflowServer(instanceID))
        workflowarchivepkg.RegisterArchivedWorkflowServiceServer(grpcServer, workflowarchive.NewWorkflowArchiveServer(wfArchive))
        clusterwftemplatepkg.RegisterClusterWorkflowTemplateServiceServer(grpcServer, clusterworkflowtemplate.NewClusterWorkflowTemplateServer())
+       reflection.Register(grpcServer)
        return grpcServer
 }
```

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by ktjooho** [comment]
This suggestion is what I really need to exactly!

**Comment by alexec** [comment]
How would you secure this API?

**Comment by alexec** [comment]
Won't fix - we will provide Swagger UI instead #3358 .
